### DISCUSSION TITLE
"You may notice some regions of the text are highlighted a faint brown within the result"

### DISCUSSION THREAD

**Discussion by endolith** [freedmand/semantra-python#59] [General]
> You may notice some regions of the text are highlighted a faint brown within the result, and those highlights stream in a bit slower. Semantra tries to explain the most relevant parts of each result to the search query. These highlights help direct your attention to what may be the most relevant.

How does this work?  Sometimes the brown snippets seem irrelevant to what is being searched for. 

I'm guessing that the total snippet shown is what was previously fed to the embedding model during ingestion, and that's what the `0.83` score refers to, and then as you scroll through those results, it does a more fine-grained search within that snippet of smaller snippets to find the brown sections?

And sometimes the brown sections don't appear at all.  What does that mean?

**Comment by freedmand** [comment]
> I'm guessing that the total snippet shown is what was previously fed to the embedding model during ingestion, and that's what the 0.83 score refers to, and then as you scroll through those results, it does a more fine-grained search within that snippet of smaller snippets to find the brown sections?

That's exactly right! It's just a more fine-grained search result. Thanks for clarifying.

> And sometimes the brown sections don't appear at all. What does that mean?

That means the code is a little buggy :) I think there's some refinement needed there I'll be taking on as part of v0.2. It probably gets bogged down because it's requesting too many snippets at once and then stalls, or is still requesting snippets for search results that are now off-screen and no longer visible.
